dodo.py

```diff
@@ -0,0 +1,325 @@
+##
+# This file contains various build steps used to make sure the data the Hedy application
+# uses is up-to-date.
+#
+# Functions that start with the name `task_` create a "task", which runs one or more
+# "actions" (shell commands or Python functions). Tasks can have file dependencies and
+# file outputs, as well as other task dependencies. If none of the files that a task
+# depends on have changed since the last time it ran, the task will be skipped. That's
+# good for speed!
+#
+# Keep the following in mind:
+# - You can specify commands as a string `'mkdir dir'` or as an array `['mkdir', 'dir']`.
+#   The second case doesn't allow shell features, but is safer in case of variables.
+# - `file_dep` requires a list of source files, but you can use `glob('*.yaml')`
```

Comment:
Every task looks like this:

```
               ┌──────────────┐                
               │              │                
input ────────▶│    Script    │────────▶ output
               │              │                
               └──────────────┘                
```

`file_dep` should at least be the set of `input` files. That way, if any of the input files change, you know that the `output` needs to be regenerated.

But what if you change the script itself? The thing that turns `input` into `output`? 

If you *don't* put the script itself into `file_dep` as well, `doit` will think that `output` is still up-to-date! Because none of the `input` files have changed, so according to the list of instructions `output` doesn't need to be regenerated.

You probably do want the task to run again if you change the script though, because probably the `output` would change. That's why you should add the scripts to `file_dep` as well.


If you change the script, you probably want to run the task again, because

Reply:
Right, that makes sense. Shouldn't we then also include the `.doit.db`? Otherwise all tasks will run no matter whether the inputs/outputs changed?